flake.nix

```diff
@@ -43,18 +44,12 @@
           '';
         };
         devShells.default =
-          with pkgs;
-          mkShell {
-            buildInputs = [
+          pkgs.mkShell {
+            packages = with pkgs; [
               bc
-              cargo
-              rustc
-              rustfmt
               pre-commit
-              rustPackages.clippy
-            ];
-            nativeBuildInputs = [ toolchain ];
-            RUST_SRC_PATH = rustPlatform.rustLibSrc;
+            ] ++ [ toolchain ];
+            RUST_SRC_PATH = toolchain.availableComponents.rust-src;
```

Comment:
This part I'm not sure about, I assume this is for VSCode integration? I don't have that or know how to check if this is still making Rust's source code available for inspection with RLS

Reply:
This seems to be necessary for rust-analyzer, for the development environment. See https://discourse.nixos.org/t/rust-src-not-found-and-other-misadventures-of-developing-rust-on-nixos/11570

Trying to run rust-analyzer from the dev environment, I get another problem though:
```
(nix:nix-shell-env) mks@Navi:~/Projects/amber$ rust-analyzer 
error: Unknown binary 'rust-analyzer' in official toolchain 'stable-x86_64-unknown-linux-gnu'.
```